src/Shared/RoslynUtils/ParsabilityHelper.cs

```diff
@@ -148,7 +148,7 @@ private static bool IsBindAsyncWithParameter(IMethodSymbol methodSymbol, ITypeSy
     private static bool IsReturningValueTaskOfTOrNullableT(INamedTypeSymbol returnType, ITypeSymbol containingType, WellKnownTypes wellKnownTypes)
     {
         return SymbolEqualityComparer.Default.Equals(returnType.ConstructedFrom, wellKnownTypes.Get(WellKnownType.System_Threading_Tasks_ValueTask_T)) &&
-            SymbolEqualityComparer.Default.Equals(returnType.TypeArguments[0].UnwrapTypeSymbol(), containingType);
```

Comment:
Does using `SymbolEqualityComparer.IncludeNullability.Equals` help here?

Reply:
My change here preserves the existing behavior following some tweaks in `UnwrapTypeSymbol(...)`. However the name of this method is actually a bit suspect since it doesn't seem to match the expression anymore. @halter73 - care to comment?

Given I'm not changing behavior here I'd prefer not to try and address refactoring this now and hold up the array support.